File: packages/telemetry/typescript/src/sdk/_deprecation.ts
/**
 * Internal helpers for one-time deprecation warnings.
 *
 * Lives in its own module so both `init.ts` and `context.ts` can depend on it
 * without forming a circular import (`init` already imports the span processor,
 * which imports `context`). Mirrors the Python `_deprecation.py` module.
 */

let projectSlugDeprecationWarned = false

/** @internal Exposed for test resets — not part of the public API. */
export function resetProjectSlugDeprecationWarningForTesting(): void {
  projectSlugDeprecationWarned = false
}

export function warnProjectSlugDeprecated(site: "constructor" | "capture"): void {
  if (projectSlugDeprecationWarned) return
  projectSlugDeprecationWarned = true
  const optionName = site === "constructor" ? "`projectSlug`" : "`projectSlug` on capture()"
  console.warn(
    `[Latitude] ${optionName} is deprecated and will be removed in a future release — rename it to \`project\`. ` +
      "Both work for now; when both are passed, `project` wins.",
  )
}

File: packages/telemetry/python/src/latitude_telemetry/sdk/_deprecation.py
"""
Internal helpers for one-time deprecation warnings.

Lives in its own module so both ``init.py`` and ``context.py`` can depend on it
without forming a circular import (``init`` already imports the span processor,
which imports ``context``).
"""

import logging

logger = logging.getLogger(__name__)

_project_slug_deprecation_warned = False


def reset_project_slug_deprecation_warning_for_testing() -> None:
    """Test-only helper to reset the once-per-process deprecation flag."""
    global _project_slug_deprecation_warned
    _project_slug_deprecation_warned = False


def warn_project_slug_deprecated(site: str) -> None:
    """Log a one-time deprecation warning for the legacy ``project_slug`` argument."""
    global _project_slug_deprecation_warned
    if _project_slug_deprecation_warned:
        return
    _project_slug_deprecation_warned = True
    option = "`project_slug`" if site == "constructor" else "`project_slug` on capture()"
    logger.warning(
        "[Latitude] %s is deprecated and will be removed in a future release — rename it to "
        "`project`. Both work for now; when both are passed, `project` wins.",
        option,
    )
